pkgs/development/libraries/kde-frameworks/kde-wrapper.nix

```diff
@@ -61,13 +61,9 @@ stdenv.mkDerivation {
         fi
     done
 
-    ln -s "$env/share" "$out"
-
+    mkdir -p "$out/nix-support"
     for drv in $unwrapped; do
-        if [ -a "$drv/nix-support/propagated-user-env-packages" ]; then
-            mkdir -p "$out/nix-support"
-            cat "$drv/nix-support/propagated-user-env-packages" >> "$out/nix-support/propagated-user-env-packages"
-        fi
+        echo "$drv" >> "$out/nix-support/propagated-user-env-packages"
```

Comment:
That's right, but my test adds _wrapped_ derivation as its path, so it creates the scenario that you describe and yet, somehow, collision doesn't occur. _Why_ it isn't is beyond me. To show it in more verbosity, this Nix expression:

```
with import <nixpkgs> {};

let
  pkg = kde5.kdeWrapper {
    unwrapped = [ kde5.gwenview.unwrapped ];
    targets = [ "bin/gwenview" ];
    paths = [ kde5.kipi-plugins ];
  };

in buildEnv {
  name = "foo";
  paths = [ pkg ];
  ignoreCollisions = false;
  checkCollisionContents = true;
}
```

should be equivalent to my older experiment (so, it won't show collision warnings). I can't reproduce it right now to be sure though because my ISP blocks `cache.nixos.org` again -_-. Can you try?